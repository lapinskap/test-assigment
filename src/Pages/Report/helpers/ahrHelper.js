export const getAhrUrl = (url, isAhr) => {
  let result = isAhr ? '/ahr' : '';
  result += url;

  return result;
};

export const asd = () => {};
