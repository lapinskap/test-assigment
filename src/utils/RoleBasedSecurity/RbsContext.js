import React from 'react';
import UserInfo from './UserInfo';

const RbsContext = React.createContext({
  isLoggedIn: false,
  login: () => {},
  logout: () => {},
  userInfo: new UserInfo({}),
});
export default RbsContext;
