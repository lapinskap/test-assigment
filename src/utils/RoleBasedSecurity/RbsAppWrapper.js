import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import RbsContext from './RbsContext';
import {
  clearSessionData, deleteSession, getSession, getUserInfo, initSession,
} from './Session';
import __ from '../Translations';
import tmpAuthorization from './tmpAuthorization';
import DefaultFallback from '../../Layout/AppMain/DefaultFallback';
import UserInfo from './UserInfo';

export default function RbsAppWrapper({ children }) {
  const [userInfo, setUserInfo] = useState({});
  const [isLoggedIn, setIsLogged] = useState(false);
  const [isInitializing, setIsInitializing] = useState(true);

  useEffect(() => {
    initializeSession();
  }, []);

  const tmpLoginAction = async (type) => {
    try {
      const {
        token, refresh_token: refreshToken, date, expiresIn, userInfo: user,
      } = await tmpAuthorization(type);

      const session = getSession();
      await session.login(token, refreshToken, date, expiresIn);
      setUserInfo(new UserInfo(user));
      setIsLogged(Boolean(await session.getValidAccessToken()));
      return null;
    } catch (e) {
      console.error(e);
      return __('Nieprawidłowy login lub hasło');
    }
  };
  const logoutAction = async () => {
    try {
      clearSessionData();
      await new Promise((resolve) => {
        window.location = `/signin/oauth/logout?redirect_url=${window.location.origin}`;
        setTimeout(() => resolve(), 5000);
      });
      return null;
    } catch (e) {
      console.error(e);
      return __('Nie udaoło się wylogować');
    }
  };

  const initializeSession = async () => {
    const session = await initSession();
    let isAlreadyLoggedIn = Boolean(await session.getValidAccessToken());
    if (isAlreadyLoggedIn) {
      try {
        const user = await getUserInfo();
        setUserInfo(new UserInfo(user));
      } catch (e) {
        clearSessionData();
        deleteSession();
        await initSession();
        isAlreadyLoggedIn = false;
      }
    }

    setIsLogged(isAlreadyLoggedIn);
    setIsInitializing(false);
  };

  if (isInitializing) {
    return DefaultFallback;
  }
  return (
    <RbsContext.Provider value={{
      isLoggedIn,
      login: tmpLoginAction,
      logout: logoutAction,
      userInfo,
    }}
    >
      {children}
    </RbsContext.Provider>
  );
}

RbsAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};
