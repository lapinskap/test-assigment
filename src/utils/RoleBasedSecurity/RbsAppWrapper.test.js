import React, { useContext } from 'react';
import { mount } from 'enzyme';
import renderer, { act } from 'react-test-renderer';
import EnzymeToJson from 'enzyme-to-json/index';
import RbsAppWrapper from './RbsAppWrapper';
import '../../tests/setupTests';
import RbsContext from './RbsContext';
import {
  clearSessionData, deleteSession, SESSION_STORAGE_KEY, USER_INFO_STORAGE_KEY,
} from './Session';
import { waitToLoadMocks } from '../../tests/helpers';

describe('<RbsAppWrapper />', () => {
  afterEach(() => {
    jest.clearAllMocks();
    clearSessionData();
    deleteSession();
  });
  it('renders correctly with children when logged', async () => {
    localStorage.setItem(SESSION_STORAGE_KEY, JSON.stringify({
      token: 'asds',
      date: Date.now(),
      expires_in: 3600,
    }));

    const wrapper = mount((
      <RbsAppWrapper>
        <div>1234</div>
      </RbsAppWrapper>
    ));
    wrapper.update();
    expect(EnzymeToJson(wrapper))
      .toMatchSnapshot();
  });
  it('renders correctly with wrong user info json', async () => {
    localStorage.setItem(SESSION_STORAGE_KEY, JSON.stringify({
      token: 'asds',
      date: Date.now(),
      expires_in: 3600,
    }));
    sessionStorage.setItem(USER_INFO_STORAGE_KEY, '{"k[]');
    const wrapper = mount((
      <RbsAppWrapper>
        <div>1234</div>
      </RbsAppWrapper>
    ));
    wrapper.update();
    expect(EnzymeToJson(wrapper))
      .toMatchSnapshot();
  });
  it('renders correctly with children when not logged', async () => {
    const tree = renderer
      .create((
        <RbsAppWrapper>
          <div>1234</div>
        </RbsAppWrapper>
      ))
      .toJSON();
    await waitToLoadMocks();
    expect(tree)
      .toMatchSnapshot();
  });
  it('should login and logout method work', async () => {
    const Content = () => {
      const { login, logout } = useContext(RbsContext);
      return (
        <div>
          <button type="button" onClick={() => login('admin', 'admin')}>Login</button>
          <button type="button" onClick={() => logout()}>Logout</button>
        </div>
      );
    };
    let wrapper;
    await act(() => {
      wrapper = mount((
        <RbsAppWrapper>
          <Content />
        </RbsAppWrapper>
      ));
    });
    wrapper.update();
    await act(() => {
      wrapper.find('button')
        .at(0)
        .prop('onClick')();
    });
    wrapper.update();
    expect(localStorage.getItem(SESSION_STORAGE_KEY)).not.toBe(null);

    await act(() => {
      wrapper.find('button')
        .at(1)
        .prop('onClick')();
    });
    expect(localStorage.getItem(SESSION_STORAGE_KEY)).toBe(null);
  });
});
