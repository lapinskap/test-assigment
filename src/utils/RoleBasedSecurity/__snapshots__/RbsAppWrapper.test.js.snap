// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RbsAppWrapper /> renders correctly with children when logged 1`] = `
<RbsAppWrapper>
  <div
    className="loader-container"
  >
    <div
      className="loader-container-inner"
    >
      <div
        className="text-center"
      >
        <Loader
          active={true}
          type="line-scale-party"
        >
          <div
            className="loader loader-undefined loader-active"
          >
            <div
              className="loader-inner line-scale-party"
            >
              <div
                key="0"
                style={Object {}}
              />
              <div
                key="1"
                style={Object {}}
              />
              <div
                key="2"
                style={Object {}}
              />
              <div
                key="3"
                style={Object {}}
              />
            </div>
          </div>
        </Loader>
      </div>
    </div>
  </div>
</RbsAppWrapper>
`;

exports[`<RbsAppWrapper /> renders correctly with children when not logged 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader-container-inner"
  >
    <div
      className="text-center"
    >
      <div
        className="loader loader-undefined loader-active"
      >
        <div
          className="loader-inner line-scale-party"
        >
          <div
            style={Object {}}
          />
          <div
            style={Object {}}
          />
          <div
            style={Object {}}
          />
          <div
            style={Object {}}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<RbsAppWrapper /> renders correctly with wrong user info json 1`] = `
<RbsAppWrapper>
  <div
    className="loader-container"
  >
    <div
      className="loader-container-inner"
    >
      <div
        className="text-center"
      >
        <Loader
          active={true}
          type="line-scale-party"
        >
          <div
            className="loader loader-undefined loader-active"
          >
            <div
              className="loader-inner line-scale-party"
            >
              <div
                key="0"
                style={Object {}}
              />
              <div
                key="1"
                style={Object {}}
              />
              <div
                key="2"
                style={Object {}}
              />
              <div
                key="3"
                style={Object {}}
              />
            </div>
          </div>
        </Loader>
      </div>
    </div>
  </div>
</RbsAppWrapper>
`;
